src/properties/forest.c

```diff
@@ -0,0 +1,215 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "igraph_structural.h"
+
+#include "igraph_adjlist.h"
+#include "igraph_constructors.h"
+#include "igraph_interface.h"
+#include "igraph_stack.h"
+
+/* count the number of vertices reachable from the root */
+static int igraph_i_is_forest_visitor(igraph_integer_t root,igraph_vector_t *visited, const igraph_adjlist_t *al, igraph_integer_t *visited_count,igraph_integer_t *res,  igraph_neimode_t mode){
+
+    igraph_stack_int_t stack;
+    igraph_integer_t i;
```

Comment:
Can you put a newline between your declarations and your code? That makes it a little easier to read.